File: joyal670-protenium-iro-hub-9d65cb9ae2c4/app/src/main/java/com/protenium/irohub/ui/main/dashboard/home/adapter/AddressAdapter.java
package com.protenium.irohub.ui.main.dashboard.home.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.RadioButton;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.protenium.irohub.R;
import com.protenium.irohub.model.getAddress.Datum;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class AddressAdapter extends RecyclerView.Adapter<AddressAdapter.ViewHold> {

    private final List<com.protenium.irohub.model.getAddress.Datum> stringList;
    private final Context context;
    private SelectedValue selectedValue;

    public AddressAdapter(Context fragment, List<com.protenium.irohub.model.getAddress.Datum> dataList) {
        this.stringList = dataList;
        this.context = fragment;
        if (fragment instanceof SelectedValue) {
            selectedValue = (SelectedValue) fragment;
        }
    }

    @NonNull
    @Override
    public ViewHold onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.recycerview_address, parent, false);
        return new ViewHold(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHold holder, int position) {

        Datum datum = stringList.get(position);

        holder.rvAddressTv.setText(stringList.get(position).getTitle() + "," +
                stringList.get(position).getGovernorate() + "," +
                stringList.get(position).getArea() + "," +
                stringList.get(position).getBlock() + "," +
                stringList.get(position).getAvenue() + "," +
                stringList.get(position).getStreet() + "," +
                stringList.get(position).getBuilding() + "," +
                stringList.get(position).getFloor() + "," +
                stringList.get(position).getAppartment());

        holder.rvAddressRadioBtn.setChecked(stringList.get(position).getChecked());

        holder.rvAddressRadioBtn.setOnClickListener(v -> {
            if (!stringList.get(position).getChecked()) {
                holder.rvAddressRadioBtn.setChecked(true);
                for (int i = 0; i < stringList.size(); i++) {
                    stringList.get(i).setChecked(false);
                }
                stringList.get(position).setChecked(true);
                notifyDataSetChanged();
                selectedValue.setValue(datum);

            }
        });

        holder.rvAddressDeleteBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                selectedValue.delete(datum);
            }
        });

    }

    @Override
    public int getItemCount() {
        return stringList.size();
    }

    public interface SelectedValue {
        void setValue(Datum datum);

        void delete(Datum datum);
    }

    public class ViewHold extends RecyclerView.ViewHolder {

        @BindView(R.id.rvAddressTv)
        TextView rvAddressTv;

        @BindView(R.id.rvAddressRadioBtn)
        RadioButton rvAddressRadioBtn;

        @BindView(R.id.rvAddressDeleteBtn)
        ImageView rvAddressDeleteBtn;


        public ViewHold(@NonNull View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);

            for (int i = 0; i < stringList.size(); i++) {
                stringList.get(i).setChecked(false);
            }

        }
    }

}

File: proteinium/app/src/main/java/com/proteinium/proteiniumdietapp/ui/main/home/home/checkout_page/adapter/AddressAdapter.kt
package com.proteinium.proteiniumdietapp.ui.main.home.home.checkout_page.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.proteinium.proteiniumdietapp.R
import com.proteinium.proteiniumdietapp.pojo.get_addresses.Address
import kotlinx.android.synthetic.main.recycerview_address.view.*
import java.util.*


class AddressAdapter(private var addressList: ArrayList<Address>,val selectedAddress : (Int,String) -> Unit, val deleteAddress : (Int,Int) -> Unit, val editAddress : (Int) -> Unit) : RecyclerView.Adapter<AddressAdapter.ViewHold>()
{

    private var selectedPosition : Int = 0
    private var context: Context? = null
    private var row_index = -1

    class ViewHold(itemView: View): RecyclerView.ViewHolder(itemView)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHold
    {
        context = parent.context
        val view = LayoutInflater.from(parent.context).inflate(R.layout.recycerview_address, parent, false)
        return ViewHold(
            view
        )
    }

    override fun getItemCount(): Int
    {
        return addressList.size
    }

    override fun onBindViewHolder(holder: ViewHold, index: Int)
    {
        //holder.itemView.rvAddressRadioBtn.isChecked = selectedPosition == position
        if(addressList[index].default==1){
            holder.itemView.rvAddressRadioBtn.isChecked=true
        }
        val address = StringBuilder()
        if(!addressList[index].governorate.isNullOrBlank()){
            address.append(context!!.getString(R.string.governorate))
            address.append(": ")
            address.append(addressList[index].governorate)
            address.append(",")
            address.append(" ")
        }

        if(!addressList[index].area.isNullOrBlank()){
            address.append(context!!.getString(R.string.area))
            address.append(": ")
            address.append(addressList[index].area)
            address.append(",")
            address.append(" ")
        }
        if(!addressList[index].block.isNullOrBlank()){
            address.append(context!!.getString(R.string.block))
            address.append(": ")
            address.append(addressList[index].block)
            address.append(",")
            address.append(" ")
        }
        if(!addressList[index].street.isNullOrBlank()){
            address.append(context!!.getString(R.string.street))
            address.append(": ")
            address.append(addressList[index].street)
            address.append(",")
            address.append(" ")
        }
        if(!addressList[index].avenue.isNullOrBlank()){
            address.append(context!!.getString(R.string.avenue))
            address.append(": ")
            address.append(addressList[index].avenue)
            address.append(",")
            address.append(" ")
        }
        if(!addressList[index].building.toString().isNullOrBlank()){
            address.append(context!!.getString(R.string.building))
            address.append(": ")
            address.append(addressList[index].building)
            address.append(",")
            address.append(" ")
        }
        if(!addressList[index].floor.toString().isNullOrBlank()){
            address.append(context!!.getString(R.string.floor))
            address.append(": ")
            address.append(addressList[index].floor)
            address.append(",")
            address.append(" ")
        }
        if(!addressList[index].appartment.toString().isNullOrBlank()){
            address.append(context!!.getString(R.string.appartment))
            address.append(": ")
            address.append(addressList[index].appartment)
        }

        holder.itemView.rvAddressTv.text = address

        holder.itemView.rvAddressRadioBtn.setOnClickListener {
            if(holder.itemView.rvAddressRadioBtn.isChecked)
            {
                row_index = index
                selectedAddress.invoke(addressList[index].id,address.toString())
                notifyDataSetChanged()
            }

        }
        if (row_index!=-1){
            holder.itemView.rvAddressRadioBtn.isChecked = row_index==index
        }
        holder.itemView.rvAddressDeleteBtn.setOnClickListener {
            deleteAddress.invoke(index,addressList[index].id)
        }

        holder.itemView.rvAddressEditBtn.setOnClickListener {
            editAddress.invoke(addressList[index].id)
        }



    }
}